### PR TITLE
Issue/886

### PR DESCRIPTION
**Pull request by lc-thomasberger**
use select 2

fixes #889

### CODE REVIEW

File: js/adapt-contrib-matching.js [hunk 1]
```diff
@@ -1,18 +1,30 @@
-define(function(require) {
-
-    var QuestionView = require('coreViews/questionView');
-    var Adapt = require('coreJS/adapt');
+define([
+    'core/js/adapt',
+    'core/js/views/questionView',
+    '../libraries/select2.js'
```
Comment (oliverfoster) [inline js/adapt-contrib-matching.js]: this should be "libraries/select2.js" did you have a problem with it?

**Comment by chris-steele** [comment]
This is working nicely. It's more accessible now that the select box can be opened by the keybaord. I just did a cross-compatibility test and only found a couple of issues (and one is pretty minor):

1. IE8 scroll jump on opening select box

matching-issue-2

2. Possible to focus select box after submission

matching-issue-1

**Comment by oliverfoster** [comment]
interesting, nice find.

**Comment by chris-steele** [comment]
The IE8 issue relates to the fact that two separate elements are receiving focus. Firstly, the `select2-selection` element within the component, then the `select2-search__field` element directly under `body`.

`jquery.a11y.js` picks up the focus events and calls `$.fn.limitedScrollTo`. Unfortunately, the position of `select2-search__field` is miscalculated and the function therefore invokes a `$.scrollTo` call, causing the jump.

It's hard to decide where the real problem is (select2, IE8, a11y), but there is a solution (albeit far from ideal). The solution I've found is to add the following check to `$.fn.limitedScrollTo`:

`if ($element.attr('class').indexOf('select2-search__field') != -1) return $element;`

after `if ($element.isFixedPostion()) return this;`

I'd appreciate your thoughts on this @lc-thomasberger, @oliverfoster, @tomgreenfield.

**Comment by oliverfoster** [comment]
i suggest you change https://github.com/adaptlearning/adapt_framework/blob/master/src/core/js/libraries/jquery.a11y.js#L350 to:
```
$element[0].getBoundingClientRect().top;
```
and remove the variable ```scrollTopWithTopOffset```

**Comment by chris-steele** [comment]
As discussed with @oliverfoster, the simplest solution to this issue seems to be appending the following to matching.less:
```
// bug fix for IE8
.select2-search__field {
	position: fixed;
}
```

@lc-thomasberger, would you mind making this change and verifying?

**Comment by lc-thomasberger** [comment]
Thanks @chris-steele and @oliverfoster implemented the css and it fixed the issue.

File: README.md [hunk 2]
```diff
@@ -97,8 +97,8 @@ contains values for three types of answers: **correct**, **_incorrect**, and **_
 No known limitations.
 
 ----------------------------
-**Version number:**  2.0.4   <a href="https://community.adaptlearning.org/" target="_blank"><img src="https://github.com/adaptlearning/documentation/blob/master/04_wiki_assets/plug-ins/images/adapt-logo-mrgn-lft.jpg" alt="adapt learning logo" align="right"></a> 
-**Framework versions:** 2.0  
+**Version number:**  2.0.5   <a href="https://community.adaptlearning.org/" target="_blank"><img src="https://github.com/adaptlearning/documentation/blob/master/04_wiki_assets/plug-ins/images/adapt-logo-mrgn-lft.jpg" alt="adapt learning logo" align="right"></a>
```
Comment (moloko) [inline README.md]: I think we should change the version to 2.1.0 (as we did for the Slider update) to reflect that this is quite a big change

File: bower.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "adapt-contrib-matching",
-  "version": "2.0.4",
-  "framework": "^2.0.11",
+  "version": "2.0.5",
```
Comment (moloko) [inline bower.json]: I think we should change the version to 2.1.0 (as we did for the Slider update) to reflect that this is quite a big change

**Comment by moloko** [comment]
all looks great, I have just three comments:
1. Should we amend the `ariaRegion` for this component in course.json to explain how to interact with it via the keyboard? It is slightly different to the previous version in that you have to press enter to open and to close it again once you have made your selection
1. It is now possible to open all three `<select>` inputs at the same time (it wasn't before)
1. On iOS, if you have more than 4/5 options, it's not at all obvious that you can scroll to see more.

I don't think 2. & 3. should prevent this from being merged (unless you particularly want to sort them now), we can deal with them separately.

**Pull request merged**